elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -139,4 +139,28 @@ default void flush() {
                                                           Optional<Sorting> sorting, Optional<Pagination> pagination) {
         return collection;
     }
+
+    default <T> Object getRelation(
+            Object entity,
+            RelationshipType relationshipType,
+            String relationName,
+            Class<T> relationClass,
+            EntityDictionary dictionary,
+            Set<Predicate> filters
+    ) {
+        return PersistentResource.getValue(entity, relationName, dictionary);
```

Comment:
I agree. Calling methods in PersistentResource from transaction feels weird. The two classes should be parallel. 

I'm not against still keeping the Dictionary tho. What's the concern of sharing dictionary around?